src/Illuminate/Bus/Dispatcher.php

```diff
@@ -239,19 +237,11 @@ public function dispatchToQueue($command)
      */
     protected function pushCommandToQueue($queue, $command)
     {
-        if (isset($command->queue, $command->delay)) {
-            return $queue->laterOn($command->queue, $command->delay, $command);
-        }
-
-        if (isset($command->queue)) {
-            return $queue->pushOn($command->queue, $command);
-        }
-
         if (isset($command->delay)) {
-            return $queue->later($command->delay, $command);
+            return $queue->later($command->delay, $command, queue: $command->queue ?? null);
```

Comment:
Is it not possible to maintain using `laterOn` and `pushOn` for `11.x` and change it for `12.x`?

Reply:
@crynobone The methods aren't going anywhere, it's just unclear if there is any implementation that does anything except wrap the other functions.

Reply:
Yeah, just thinking if existing application using `Bus::spy()` and `Bus::shouldHaveReceived('laterOn')` and with these changes, it starts to cause failing tests.

Reply:
Happy to back that part out, although those feel like brittle tests I don't think it's good to break things for the sake of breaking things.